clash-prelude/tests/Clash/Tests/Resize.hs

```diff
@@ -49,5 +198,16 @@ tests = testGroup "Resize"
     [ testProperty "indexProp @17 @19" (indexProp @17 @19 Proxy)
     , testProperty "indexProp @19 @19" (indexProp @19 @19 Proxy)
     , testProperty "indexFailProp @37 @7" (indexFailProp @37 @7 Proxy)
+    , testPropertyNamed "maybeResizeUnsignedProp" "maybeResizeUnsignedProp" maybeResizeUnsignedProp
+    , testPropertyNamed "maybeResizeSignedProp" "maybeResizeSignedProp" maybeResizeSignedProp
+    , testPropertyNamed "maybeResizeIndexProp" "maybeResizeIndexProp" maybeResizeIndexProp
+    , testPropertyNamed "maybeResizeBitVectorProp" "maybeResizeBitVectorProp" maybeResizeBitVectorProp
+    , testPropertyNamed "maybeTruncateUnsignedProp" "maybeTruncateUnsignedProp" maybeTruncateUnsignedProp
+    , testPropertyNamed "maybeTruncateSignedProp" "maybeTruncateSignedProp" maybeTruncateSignedProp
+    , testPropertyNamed "maybeTruncateIndexProp" "maybeTruncateIndexProp" maybeTruncateIndexProp
+    , testPropertyNamed "maybeTruncateBitVectorProp" "maybeTruncateBitVectorProp" maybeTruncateBitVectorProp
     ]
   ]
+
+main :: IO ()
+main = defaultMain $ adjustOption (\_ -> HedgehogTestLimit (Just 1_000_000)) $ tests
```

Comment:
Did you intend to include this in the PR or was it just for local testing?

Reply:
Both! I'll make it less opinionated, but it's really nice for test modules to have `main`s. Then I can do:

```
./repld p:tests -W -T Clash.Tests.Resize.main
```

Reply:
... _Or_ ...

We extend `repld` to allow passing [command line arguments](https://github.com/ndmitchell/ghcid?tab=readme-ov-file#how-do-i-run-pass-command-arguments-with---test) to `clash-prelude/tests/unittests.hs`, including arbitrrary Tasty options. You could do

```
./repld p:tests -W -T unittests -S Resize -A "--hedgehog-tests 1000000"
```
(S for _specific test_ and A for _additional arguments_)
and it would invoke
```
ghcid ... -T Main.main --setup ":set args -p Resize --hedgehog-tests 1000000"
```

Reply:
Oh huh, I didn't know you could do that. Are you insisting on removing the `main`? I don't really mind it existing, even if the same thing can be achieved in another way.

Reply:
Oh no, I just wondered if there wasn't a "more nice" way to achieve the result, and this seemed like one. Go ahead and keep it as is.